### PR TITLE
[Login] Unit tests

### PR DESCRIPTION
**Pull request by hitesh96db**

### CODE REVIEW

**Comment by hitesh96db** [comment]
Initially I had a doubt https://github.com/privly/privly-applications/pull/184#issuecomment-77741676,
But since these are unit tests I can assume that the ajax calls will work, and I can test based on a fake response object ?

**Comment by smcgregor** [comment]
Do you still want me to review this PR after reading my last comment on https://github.com/privly/privly-applications/pull/184 ?

**Comment by hitesh96db** [comment]
@smcgregor I think I did do something similar as you mentioned, Could you review this and tell me if I am heading in the right direction ?

**Comment by smcgregor** [comment]
You are heading in the right direction, but you still need to isolate the functions better in each test. They are calling each other in a cascading fashion. Additionally, it is good to work towards tests that only test one thing at a time and are short.

**Comment by hitesh96db** [comment]
@smcgregor Updated the tests. Please review :)

**Comment by hitesh96db** [comment]
@smcgregor Please review :)

**Comment by smcgregor** [comment]
Can you remove the `var keys = Object.keys(__html__);` section and the `beforeEach`? Does it still pass?

File: Login/test/new.js [hunk 1]
```diff
@@ -20,18 +20,44 @@ describe ("Login Suite", function() {
...
     $("#user_password").val("danger.dont.use.bork.bork.bork");
-    
-    $("#login").click();
   });
-  
+
+  it("tests submitCredentials", function() {    
+    spyOn(privlyNetworkService, "sameOriginPostRequest").and.callFake(function(url, callback, data) {});
+    callbacks.submitCredentials();
+    expect($("#login").prop("disabled")).toBe(true);
```
Comment (hitesh96db) [inline Login/test/new.js]: I need Object.keys(__html__) and beforeEach for this.

Comment (smcgregor) [inline Login/test/new.js]: Create the expected DOM element in JS so you don't need to load the entire HTML page.

Comment (hitesh96db) [inline Login/test/new.js]: :+1:

**Comment by hitesh96db** [comment]
@smcgregor Done.

**Comment by smcgregor** [comment]
[Running on TravisCI](https://travis-ci.org/smcgregor/privly-applications/builds/53828035)

**Pull request merged**